### DISCUSSION TITLE
your focus-trap must have at least one container with at least one tabbable node focus-trap

### DISCUSSION THREAD

**Discussion by NicolasLaverde** [focus-trap/focus-trap#896] [Q&A] [answered]
Hi fellas, 
I have some weird bug, for some extrange reason we get this bug in just one client is implementing our service ( some Cookie consent modal ) 
This client is getting this bug error => 
![Screenshot 2023-02-13 at 11 11 25 AM](https://user-images.githubusercontent.com/12724568/218511078-111b6d5c-5892-4211-8463-2f4a9fb0cc4d.png)

But when I open their web Page I can see this error at all. This happens to us several times. 
What should be? Chrome or firefox version? why this can works in many clients in the script we create but not in 100% of them? 

What's the cause of this error?

**Comment by stefcameron** [comment, marked as answer]
@NicolasLaverde Many people run into this error, unfortunately. It happens when the containers you gave when you [setup](https://github.com/focus-trap/focus-trap#createfocustrapelement-createoptions) your focus trap do not have -- or no longer have -- any tabbable elements in them, which means that focus will inevitably escape your trap because focus __must__ always go _somewhere_.

If your trap does not have (or no longer has) any tabbable (and therefore focusable) elements in it, and it was not configured with a backup element (see the `fallbackFocus` [option](https://github.com/focus-trap/focus-trap#createoptions) -- which must still be in the trap, but does not necessarily have to be tabbable (i.e. it has `tabindex="-1"`, making it focusable, but not tabbable), you will get this error.

This happens often when people have traps related to elements that appear and disappear dynamically. Typically, the error will fire either as the element is being shown (because the trap gets created before the trapped children have been inserted into the DOM), or as it's being hidden (because the trapped children are destroyed before the trap is either destroyed or disabled). I have yet to see it related to browser vendors or versions.

**Reply by NicolasLaverde** [reply]
Thanks so much ! yeah you just need to include focusTrapOptions object

**Reply by stefcameron** [reply]
Glad it helped. I decided to add a [section](https://github.com/focus-trap/focus-trap#error-your-focus-trap-must-have-at-least-one-container-with-at-least-one-tabbable-node-in-it-at-all-times) in the README about it.
